std/traits.d

```diff
@@ -8061,3 +8062,2432 @@ enum isCopyable(S) = is(typeof(
+        void bar1(string s) @safe pure @nogc nothrow { }
+    }
+
+    FooBar foobar = new FooBar;
+    int z2 = foobar.foo1(1, "");
+    double z3 = foobar.foo1("");
+    string z4 = foobar.bar1(1);
+    foobar.bar1("");
+
+    static assert(doBothHaveMember!(FooBar, Bar, "bar1"));
+    static assert(doBothHaveMember!(Bar, FooBar, "bar1"));
+}
+
+//returns true if subtype, false if there is something that does not match
+//interface
+private bool checkSubTypeOf(alias S, alias T)()
+    if (isAggregateType!S && isAggregateType!T)
+{
+    bool result;
+
+    foreach (member; __traits(allMembers, T))
+    {
+        static if (!isMemberOfObject!(T, member) &&
+                   memberCompiles!(T, member))
+        {
+            result = doBothHaveMember!(S, T, member);
+            if (result == false)
+                break;
+        }
+    }
+    return result;
+}
+
+/++
+ + Detect whether type `S` is a subtype of type `T`
+ +
+ + If type `S` implements all the members, member functions, or aggregate
+ + types of `T`, then for the sake of this function, it is said to
+ + subtype `T`. `S` does not need to explicitly inherit from `T` to subtype.
+ +
+ + This template ignores private members and/or member functions.
+ +
+ + It also ignores members and/or member functions that come from
+ + $(LREF Object). The implication of this is that a struct can be a subtype
+ + of a class, even if it does not implement $(LREF Object)'s member functions.
+ +
+ + This function ignores UDAs, but does consider D's built-in function
+ + attributes in the following manner (< indicates subtypes):
+ +      1) @safe < @trusted < @system
+ +      2) @nogc < non-@nogc (because there is no @gc)
+ +      3) nothrow < non-nothrow (because there is no throws)
+ +      4) pure < non-pure (because there is no impure)
+ +
+ + Further, it also considers the abstract/final status of member functions.
+ +      1) final < non-final
+ +      2) non-abstract < abstract
+ +
+ + Params:
+ +      S = type to be tested
+ +      T = type to be tested against
+ +
+ + Returns:
+ +      true if S is a subtype, false otherwise
+ +
+ + See_Also:
+ +      $(HTTP https://en.wikipedia.org/wiki/Subtyping, Subtyping)
+ +/
+template isSubtypeOf(alias S, alias T)
+    if (isAggregateType!S && isAggregateType!T)
+{
+    static if (checkSubTypeOf!(S, T))
+        enum bool isSubtypeOf = true;
+    else
+        enum bool isSubtypeOf = false;
+}
+
+///
+@safe pure
+unittest
+{
+    class Foo
+    {
+        int foo1(int i, string s) @safe @nogc pure nothrow { return 0; }
+        double foo1(string s) @safe @nogc pure nothrow { return 0; }
+    }
+
+    Foo foo = new Foo;
+    int z1 = foo.foo1(1, "");
+    double z2 = foo.foo1("");
+
+    struct Bar
+    {
+        string bar1(double d) @safe @nogc pure nothrow { return ""; }
+        void bar1(string s) @safe @nogc pure nothrow { }
+    }
+
+    Bar bar;
+    string z3 = bar.bar1(1);
+    bar.bar1("");
```

Comment:
please remove any unused variables or function calls

Reply:
The way the CI works is that it counts coverage based on all functions, including those in unittests. So when I add structs with member functions that I only intend to test whether they exist at compile-time, the CI will say that the unittest coverage is below what is required unless I add those. The only way I could pass the CI was including them. 

Perhaps another way to look at it is that the compiler might optimize this stuff away unless I actually provide some sort of implementation.